### ISSUE TITLE
psc-bundle: "the module could not be parsed"

### ISSUE THREAD

**Issue by hdgarrood** [purescript/purescript#1233]
For example, after running `psc`:

```
$ psc-bundle output/*/index.js --module Benchmark.Main --main Benchmark.Main | node
The module could not be parsed.
```

Is it obvious what I'm doing wrong? Also it would be nice to say which module it was that couldn't be parsed, and what it was trying to parse it as.

**Comment by paf31** [comment]
Let's print the parse error from `language-javascript` if we can. Are there any modern features being used in your module which might not be supported?

Another option, if the module can't be parsed, is to include it verbatim, but that means we can't reliably perform dead code elimination.

**Comment by hdgarrood** [comment]
This is modern javascript features in FFI files, yes? I don't think so, but this project has a lot of dependencies (too many to go through manually).

I think being able to produce a bundle with incomplete DCE is much better than producing nothing at all. In many cases I don't care about the size of the output. Perhaps we could print a warning along the lines of this:


> Warning: the following modules could not be parsed. This means that dead code elimination has not been performed on them:
> * blah/blah/FFI.js: error on line 1, col 1, expected "{", got "=>"...
> * ...

**Comment by paf31** [comment]
Well, `browserify` might be an alternative in that case?

**Comment by hdgarrood** [comment]
Ah, good point. So would you rather `psc-bundle` produces nothing and exits nonzero if this happens?

**Comment by hdgarrood** [comment]
where "this" = parse error inside `language-javascript`

**Comment by paf31** [comment]
For now, I think so, yes, but possibly with a better error message.

**Comment by garyb** [comment]
I think I'd prefer `psc-bundle` to still work but drop the unparseable code in verbatim - some DCE is better than no DCE at all, if you're running `psc-bundle` for that. That's what I'd be planning on when we were going to do it in the old style `psc`.

**Comment by paf31** [comment]
But if you can't parse the module, you can't determine its dependencies. Maybe it's a good reason to support annotations?

**Comment by hdgarrood** [comment]
So in order to be able to safely do any DCE at all, you have to be able to parse every single module, in order to determine dependencies? "Some DCE" isn't an option, it's all or none, is that right?

**Comment by paf31** [comment]
With the current approach to parsing, yes.

**Comment by garyb** [comment]
Hm, good point, I hadn't thought of that. It previously would have worked as we could have used the info from the PS AST.

**Comment by hdgarrood** [comment]
Is this fixed by #1307?

**Comment by paf31** [comment]
Yes, I think this can be considered fixed, now that we have better error messages.
